src/ios/ListLibraryItems.m

```diff
@@ -320,16 +318,21 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didComp
         NSString * errorMessage;
         
         if (error) {
+            NSLog(@"--- STATUS NOK ---");
             NSLog(@"Error: %@", error);
             errorMessage = [error localizedDescription];
         }
         else {
             errorMessage = [NSHTTPURLResponse localizedStringForStatusCode:status];
         }
         NSLog(@"Error: %@", error);
-        NSMutableDictionary * json = [NSMutableDictionary dictionaryWithObjects:@[[NSString stringWithFormat:@"%ld", status], errorMessage, [mLocalTempURL absoluteString], target]
-                                                                        forKeys:@[@"code", @"message", @"source", @"target"]];
-        [self returnUploadResult:NO payload:json command:mCommand];
+        if(mLocalTempURL == nil) {
+            NSLog(@"The mLocalTempURL is nil, the app has been killed during upload.");
```

Comment:
Instead of having a log that explains what is going on, maybe we can make the code more explicit? Something like

```
bool taskCanceledDueToAppKill = mLocalTempURL == nil && [error code] == NSTaskCanceled]

if (!taskCanceledDueToAppKill) {
  NSMutableDictionary * json etc…
  [self returnUploadResult:NO payload:json command:mCommand];
}
```

(pseudo code, i didn't test it). What do you think?

Reply:
Understood, I guess this is a great idea and using this method we can really know as you said if the task is canceled due to app kill.